scalyr_agent/scalyr_client.py

```diff
@@ -310,7 +310,7 @@ def __send_request(
                 else:
                     body_str = body
             else:
-                body_str = ""
+                body_str = b""
```

Comment:
Good catch :+1: 

The decode below would indeed fail if we don't ensure this variable contains bytes.